### PR TITLE
Respond with correct error response codes, and don't fail if the oembed response is missing a title.

### PR DESCRIPTION
**Pull request by samgiles**
Handle errors correctly in the final express handler, don't error if there is a falsey, or missing title from the oembed response.

@arcturus r?

### CODE REVIEW

**Comment by wilsonpage** [comment]
We could do with some unit-tests wrapped round this logic.

File: lib/routes/api/metadata/processor.js [hunk 1]
```diff
@@ -43,9 +43,6 @@ module.exports = function(url, options) {
       function done(json) {
         json = json || {};
         debug('parsed', json);
-        if (!json.title) {
-          return Promise.reject(new Error('empty'));
-        }
```
Comment (wilsonpage) [inline lib/routes/api/metadata/processor.js:48]: @samgiles why was this removed?

Comment (wilsonpage) [inline lib/routes/api/metadata/processor.js:48]: This will reintroduce bugs on the client when URLs don't have any content.

Comment (samgiles) [inline lib/routes/api/metadata/processor.js:48]: It's only the 'title' field? Should the client be depending on a non-empty title?

It was removed because sched was returning an empty string for the title, so the response was an error - where is the title used on the client?
